src/pyadjoint/adjoint_source.py

```diff
@@ -140,6 +146,41 @@ def _write_specfem(self, buf, time_offset):
 
         np.savetxt(buf, to_write)
 
+    def write_to_asdf(self, ds, time_offset):
+          """
+          Writes the adjoint source to an ASDF file.
+          Note: For now it is assumed SPECFEM will be using the adjoint source
+
+          :param ds: The ASDF data structure read in using pyasdf
+          :type ds: str
+          """
+          # Convert the adjoint source to SPECFEM format
+          l = len(self.adjoint_source)
+          specfem_adj_source = np.empty((l,2))
+          specfem_adj_source[:, 0] = np.linspace(0, (l - 1) * self.dt, l)
+          specfem_adj_source[:, 1] += time_offset
+          specfem_adj_source[:, 1] += self.adjoint_source[::-1]
+
+          tag = "%s_%s_%s" % (self.network, self.station, self.component)
+          min_period = self.min_period
+          max_period = self.max_period
+          component = self.component
+          station_id = "%s.%s..%s" % (self.network, self.station, self.component)
+          latitude = ds.waveforms.self.network_self.station.coordinates['latitude']
+          longitude = ds.waveforms.self.network_self.station.coordinates['longitude']
+          elevation_in_m = ds.waveforms.self.network_self.station.coordinates['elevation_in_m']
```

Comment:
For the coordinates: It might be useful to also be able to specify them as parameters to the `write_to_asdf()` method. If they are given, use them. If not try to get them from the ASDF file. If that fails raise an error.